sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1363,7 +1416,11 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
           if ($c.toByte($wrapper)) {
             $evPrim = (byte) $wrapper.value;
           } else {
-            $evNull = true;
+            if ($ansiEnabled) {
```

Comment:
instead of duplicating the code, how about we introduce a `UTF8String.toIntExact`? then both interpreted code path and codegen can call it.

Reply:
You mean something like this ->
In UTF8String.java
```
public boolean toIntExact(Boolean ansiEnabled) {
  if (ansiEnabled) {
    throw new NumberFormatException("invalid input syntax for type numeric: $c");
  }
  return true;
}
```
and here 
`$evNull = UTF8String.toIntExact`